File: py_c_java_fib/Fib.java
/**
 * 
 */

/**
 * @author raf
 *
 */
public class Fib {

    public static void main(String[] args) {
        long res;
        int n = 29;
        
        long startTime = System.nanoTime();
        res = fib(n);
        double elapsedTime = (System.nanoTime() - startTime)/1e9;

        System.out.println("fib(" + n + ") = " + res);
        System.out.println("elapsed time = " + elapsedTime + " s");
    }

    private static long fib(int n) {
        if (n == 0) 
            return 0l;
        else if (n == 1)
            return 1l;
        else
            return fib(n-1) + fib(n-2);
    }
}

File: py_c_java_fib/fib.py
# -*- coding: utf-8 -*-
"""
Created on Sun Feb 19 11:45:05 2017

@author: raf
"""
from time import time

def fib(n):
    if n == 0: return 0
    elif n == 1: return 1
    else: return fib(n-1)+fib(n-2)

n = 29
t0 = time()
res = fib(n)
et = (time() - t0)
print('fib({}) = {}'.format(n, res))
print('elapsed time = {} s'.format(et))
